update google-cloud-bigquery dependency range

tswast opened the pull request:
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Towards ibis-project/ibis#2413

tswast committed:
update google-cloud-bigquery dependency range

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * ``requirements: run: protobuf>=3.12.0`` must contain a space between the name and the pin, i.e. ``protobuf >=3.12.0``

tswast committed:
remove transitive protobuf dep

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

tswast commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.20.3, conda-smithy 3.7.10, and conda-forge-pinning 2020.10.01.12.48.12

tswast committed:
re-add protobuf dep

3.11.0 is incompatible with latest google-cloud-bigquery

jreback commented on recipe/meta.yaml:
```diff
@@ -26,15 +26,14 @@ requirements:
-    # see https://github.com/googleapis/python-api-core/issues/47
-    - google-api-core =1.17.0
-    - google-cloud-bigquery >=1.0.0
+    - google-cloud-bigquery >=1.12.0,<2.0.0dev
```
perfect!

jreback commented:
thanks @tswast ping me when this is green (or at least *some* of the builds work lol)

tswast committed:
use google-cloud-bigquery-core to avoid pyarrow conflict

Package pyarrow conflicts for:
pymapd=0.23.0 -> pyarrow==0.15.0
pyspark[version='>=2.4.3'] -> pyarrow[version='>=0.15.1']
google-cloud-bigquery[version='>=1.12.0,<2.0.0dev'] -> pyarrow[version='>=0.13.0']
pyarrow[version='>=0.15']

tswast committed:
cap google-cloud-bigquery before 1.24.0

I believe version 1.24.0 is the first that required newer versions of pyarrow, which conflicts with pymapd.

tswast committed:
temporarily reverting omni move to see if I can get a green build

tswast commented:
@jreback All green!

Might need to revert 09af52b9b52b89fb5d87b7260f5a8720ec4dce70 though based on my understanding of https://github.com/conda-forge/ibis-framework-feedstock/pull/47

jreback commented:
thanks @tswast
